node/pkg/db/governor.go

```diff
@@ -66,7 +67,14 @@ func (t *Transfer) Marshal() ([]byte, error) {
 	buf.Write(t.OriginAddress[:])
 	vaa.MustWrite(buf, binary.BigEndian, t.EmitterChain)
 	buf.Write(t.EmitterAddress[:])
-	buf.Write([]byte(t.MsgID))
+	vaa.MustWrite(buf, binary.BigEndian, uint16(len(t.MsgID)))
+	if len(t.MsgID) > 0 {
+		buf.Write([]byte(t.MsgID))
```

Comment:
Why is the write conditional here when it wasn't before?  I get why the new parm (Hash) is conditional.

Reply:
I just think it's safer, and it doesn't hurt anything.